test/integration/channel-transfer_test.go

```diff
@@ -0,0 +1,733 @@
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := conn.Close()
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		By("creating channel transfer API client")
+		apiClient = cligrpc.NewAPIClient(conn)
+
+		By("creating channel transfer request")
+		transferID := uuid.NewString()
+		channelTransferArgs := []string{transferID, "CC", "FIAT", "250"}
+
+		requestID := uuid.NewString()
+		nonce := client.NewNonceByTime().Get()
+		signArgs := append(append([]string{"channelTransferByCustomer", requestID, cmn.ChannelFiat, cmn.ChannelFiat}, channelTransferArgs...), nonce)
+		publicKey, sign, err := user.Sign(signArgs...)
+		Expect(err).NotTo(HaveOccurred())
+
+		transfer := &cligrpc.TransferBeginCustomerRequest{
+			Generals: &cligrpc.GeneralParams{
+				MethodName: "channelTransferByCustomer",
+				RequestId:  requestID,
+				Chaincode:  cmn.ChannelFiat,
+				Channel:    cmn.ChannelFiat,
+				Nonce:      nonce,
+				PublicKey:  publicKey,
+				Sign:       base58.Encode(sign),
+			},
+			IdTransfer: channelTransferArgs[0],
+			ChannelTo:  channelTransferArgs[1],
+			Token:      channelTransferArgs[2],
+			Amount:     channelTransferArgs[3],
+		}
+
+		By("sending transfer request")
+		r, err := apiClient.TransferByCustomer(clientCtx, transfer)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(r.Status).To(Equal(cligrpc.TransferStatusResponse_STATUS_IN_PROCESS))
+
+		By("checking transfer status")
+		transferStatusRequest := &cligrpc.TransferStatusRequest{
+			IdTransfer: transferID,
+		}
+
+		excludeStatus := cligrpc.TransferStatusResponse_STATUS_IN_PROCESS.String()
+		value, err := anypb.New(wrapperspb.String(excludeStatus))
+		Expect(err).NotTo(HaveOccurred())
+
+		transferStatusRequest.Options = append(transferStatusRequest.Options, &typepb.Option{
+			Name:  "excludeStatus",
+			Value: value,
+		})
+
+		ctx, cancel := context.WithTimeout(clientCtx, network.EventuallyTimeout*2)
+		defer cancel()
+
+		By("awaiting for channel transfer to respond")
+		statusResponse, err := apiClient.TransferStatus(ctx, transferStatusRequest)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(statusResponse.Status).To(Equal(cligrpc.TransferStatusResponse_STATUS_COMPLETED))
+
+		By("checking result balances")
+		client.Query(network, peer, cmn.ChannelFiat, cmn.ChannelFiat,
+			checkResult(checkBalance("750"), nil),
+			"balanceOf", user.AddressBase58Check)
+
+		client.Query(network, peer, cmn.ChannelCC, cmn.ChannelCC,
+			checkResult(checkBalance("250"), nil),
+			"allowedBalanceOf", user.AddressBase58Check, "FIAT")
+	})
+
+	It("transfer status with wrong transfer id test", func() {
+		By("creating grpc connection")
+		clientCtx = metadata.NewOutgoingContext(context.Background(), metadata.Pairs("authorization", networkFound.ChannelTransfer.AccessToken))
+
+		transportCredentials := insecure.NewCredentials()
+		grpcAddress := networkFound.ChannelTransfer.HostAddress + ":" + strconv.FormatUint(uint64(networkFound.ChannelTransfer.Ports[cmn.GrpcPort]), 10)
```

Comment:
By any chance is there no function in go that will do this better and more correctly than we can?